src/Sharpliner.Model/Job.cs

```diff
@@ -3,20 +3,40 @@
 
 namespace Sharpliner.Model
 {
-    public record Job(
-        string Name,
-        string DisplayName,
-        IEnumerable<string> DependsOn,
-        string Condition,
-        JobStrategy Strategy,
-        JobPool Pool,
-        ContainerReference Container,
-        TimeSpan Timeout,
-        TimeSpan CancelTimeout,
-        IEnumerable<Variable> Variables,
-        IEnumerable<Step> Steps,
-        JobWorkspace Workspace = JobWorkspace.Outputs,
-        bool ContinueOnError = false);
 
     // TODO: missing properties Uses, Services
+    public record Job
+    {
+        public string Name { get; }
+
+        public string DisplayName { get; }
+
+        public IEnumerable<string>? DependsOn { get; init; }
+
+        public string? Condition { get; init; }
+
+        public JobStrategy? Strategy { get; init; }
+
+        public Pool? Pool { get; init; }
+
+        public ContainerReference? Container { get; init; }
+
+        public TimeSpan? Timeout { get; init; }
+
+        public TimeSpan? CancelTimeout { get; init; }
+
+        public IEnumerable<Variable>? Variables { get; init; }
+
+        public IEnumerable<Step>? Steps { get; init; }
+
+        public JobWorkspace Workspace { get; init; } = JobWorkspace.Outputs;
+
+        public bool ContinueOnError { get; init; } = false;
+
+        public Job(string name, string displayName)
+        {
+            Name = name ?? throw new ArgumentNullException(nameof(name));
```

Comment:
So we need that with nullable? I mean, with and older version of C# the API is horrible...

Reply:
If someone who doesn't use nullable uses our library, he can pass null in. Or someone else could make a mistake and pass null even when they have nullability on.

I am not 100% sure display name is actually needed by AzDO but maybe we should require it?